custom_components/unifiprotect/select.py

```diff
@@ -242,16 +202,47 @@ def __init__(
         protect_data: UnifiProtectData,
         device: ProtectAdoptableDeviceModel,
         description: UnifiProtectSelectEntityDescription,
-        options: list[dict[str, Any]],
+        options: list[dict[str, Any]] | None,
     ):
         """Initialize the unifi protect select entity."""
-        super().__init__(protect, protect_data, device, description)
         self.device: Camera | Viewer | Light = device
+        super().__init__(protect, protect_data, device, description)
         self._attr_name = f"{self.device.name} {self.entity_description.name}"
-        self._attr_options = [item["name"] for item in options]
         self._data_key = description.ufp_value
-        self._hass_to_unifi_options = {item["name"]: item["id"] for item in options}
-        self._unifi_to_hass_options = {item["id"]: item["name"] for item in options}
+
+        if options is not None:
+            self._attr_options = [item["name"] for item in options]
+            self._hass_to_unifi_options = {item["name"]: item["id"] for item in options}
+            self._unifi_to_hass_options = {item["id"]: item["name"] for item in options}
+        self._async_update_device_from_protect()
+
+    @callback
+    def _async_update_device_from_protect(self):
+        super()._async_update_device_from_protect()
+
+        if self.entity_description.key in (_KEY_VIEWER, _KEY_DOORBELL_TEXT):
+            if self.entity_description.key == _KEY_VIEWER:
+                options = [
+                    {"id": item.id, "name": item.name}
+                    for item in self.protect.bootstrap.liveviews.values()
+                ]
+            elif self.entity_description.key == _KEY_DOORBELL_TEXT:
+                default_message = (
+                    self.protect.bootstrap.nvr.doorbell_settings.default_message_text
+                )
+                messages = self.protect.bootstrap.nvr.doorbell_settings.all_messages
+                built_messages = (
+                    {"id": item.type.value, "name": item.text} for item in messages
+                )
+
+                options = [
+                    {"id": "", "name": f"Default Message ({default_message})"},
+                    *built_messages,
+                ]
+
+            self._attr_options = [item["name"] for item in options]
+            self._hass_to_unifi_options = {item["name"]: item["id"] for item in options}
+            self._unifi_to_hass_options = {item["id"]: item["name"] for item in options}
```

Comment:
Options aren't allowed to change after the entity has been initialized since integrations that consume them once (ie Alex, HomeKit, Google Assistant, etc) will break

Reply:
Is there a better way to handle this then?  Liveviews and the Doorbell texts are by their nature very dynamic, and it is rather annoying to have to reload UFP/restart HA to get the new options.

Reply:
I can't think of a better way right now.  I agree it is annoying, but it is what we have right now.  It needs to be improved int he base architecture

Reply:
Persistent notification to alert the user they have changed, and they need to reload the integration maybe?

Reply:
Persistent notifications should only be shown for high importance items that the user is currently interacting with or need to take action soon so that seems a bit too much.

For other integrations we have logged an error that the sources/data is out of sync and a restart is required to get everything back in sync.